### ISSUE TITLE
Many BBEdit menu commands not available in Finbar

### ISSUE THREAD

**Issue by ecormany** [roeybiran/finbar-issues#11]
Most BBEdit menu commands are listed in Finbar as disabled. Selecting them plays the alert sound. All commands I've tried are accessible using the macOS Help menu. I haven't been able to discern a pattern for which commands are unavailable. 

- All system-wide standard commands (e.g., Save, Close Window, Find…) are enabled. 
- Some commands that are clearly unique to BBEdit, like Hex Dump File are available (but Hex Dump Front Window, which is right below it in the File menu isn't!)
- I thought it could be related to commands that change when holding the option key (many commands remove the ellipsis at the end and become immediate actions) but Remove Blank Lines doesn't change and is unavailable.
- I thought it could be related to the ⭐ that BBEdit lists next to paid features, but that also seems to have no bearing on which commands are available in Finbar.

I hope there's a way to fix this. BBEdit is so feature-packed with menu commands that it would be perfect to control with Finbar.
<img width="787" alt="Screenshot 2023-08-03 at 10 05 16 AM" src="https://github.com/roeybiran/finbar-issues/assets/1958332/95a6ae1b-40af-44de-b308-95bcf7d2fcda">
<img width="406" alt="Screenshot 2023-08-03 at 10 05 29 AM" src="https://github.com/roeybiran/finbar-issues/assets/1958332/a101b0de-a71c-4ebb-b7f1-fbbd3834d6dc">
<img width="787" alt="Screenshot 2023-08-03 at 10 08 25 AM" src="https://github.com/roeybiran/finbar-issues/assets/1958332/5aca62ac-fef8-4c22-8ed3-b5d24f34cfbc">

**Comment by roeybiran** [comment]
Hi Ed!

I did some poking around and it seems that BBEdit is extremely vigilant about whether its truly the focused app or not.

Notice the color of the controls in while actively using BBEdit:

<img width="862" alt="Screenshot 2023-08-04 at 11 39 40" src="https://github.com/roeybiran/finbar-issues/assets/37002381/c96c0044-dbaa-4441-a5f4-6b136c196344">

And when Finbar is activated:

<img width="868" alt="Screenshot 2023-08-04 at 11 50 40" src="https://github.com/roeybiran/finbar-issues/assets/37002381/75668efb-3de3-4489-b5dc-43b8b07f11e4">

They’re slightly dimmed.

I guess BBEdit sees that it’s not really the focused app and dynamically disables any menu items that require this context; or perhaps when focus is lost then text selection is invalidated, consequently disabling any menu items that depend on that (like “Process Lines”). To be honest, this is quite a peculiar behavior I haven’t seen in other text editors and it would be best to consult BBEdit’s developers about this, but I wouldn’t want to bother them with quirks related to my little app. However, BBEdit is such a canonical app that I’d make every effort for it to be supported properly by Finbar.

When I tell Finbar to ignore an item’s disabled state, selection works well. Here are my suggestions:

1. Pressing down a modifier key while executing an item temporarily overrides its disabled state.
2. Introduce a setting that ignores item’s disabled state for a given app. The upside of this compared to the first approach is that if an item is disabled, it won’t affect the sorting stage.

Please tell me what you think!

P. S. - I checked, and this happens with Finbar’s competitors as well — the items don’t even appear.

**Comment by ecormany** [comment]
Interesting! I dug through BBEdit's excellent documentation of their [super-advanced preferences](https://www.barebones.com/support/bbedit/ExpertPreferences.html) only accessible by `defaults write` and didn't turn up anything that could help on that side.

I think of your two proposals, I like :two: better.

**Comment by roeybiran** [comment]
Great! Do you think this option should be a part of the rule editor, or should it reside in a section of its own? 

I’m partial to the latter, as the rule editor deals with a menu item’s “intrinsic” properties (title, depth…), and our goal here is to deliberately override them in a per-app basis.

Looking forward to your input!

**Comment by roeybiran** [comment]
After some research I discovered that BBEdit is not the only app that behaves this way, and I think some apps exhibit this behavior partially, to some of their menu items.

So, going forward and in the Finbar update released today, I decided to simply let users select disabled menu items. If an item is “truly” disabled, Finbar will notify you. [Read more here](https://www.roeybiran.com/apps/finbar/help#disabled-menu-items).

If you have any suggestion to introduce settings to control the behavior of disabled menu (dimmed or not, sorted differently etc.) items in Finbar, I’d love to hear them.
